validation/validate.py

```diff
@@ -0,0 +1,215 @@
+import yaml
+import pytest
+import re
+from typing import Any
+
+
+def read_yaml_file(file_name: str) -> str:
+    '''
+    Reads a YAML file and returns it as a dictionary.
+    '''
+    with open(file_name, 'r') as f:
+        try:
+            d = yaml.safe_load(f)
+        except yaml.YAMLError as e:
+            print(e)
+    return d
+
+
+def type_matches(value: Any, expected_type: str) -> bool:
+    '''
+    Checks whether value has the type expected_type.
+    If yes, returns True, otherwise False.
+
+    Allowed types T are: 'str', 'int', 'float', 'complex', 'bool',
+                         as well as 'List[T]'.
+
+    For any other types the function raises ValueError.
+    '''
+    allowed_basic_types = ['str', 'int', 'float', 'complex', 'bool']
+    allowed_list_types = ['List[{}]'.format(t) for t in allowed_basic_types]
+    allowed_types = allowed_basic_types + allowed_list_types
+
+    # first verify whether expected_type is allowed
+    if expected_type not in allowed_types:
+        raise ValueError('could not recognize expected_type: {}'.format(expected_type))
+
+    expected_type_is_list = re.match(r"^List\[\w+\]$", expected_type) is not None
+
+    if expected_type_is_list:
+        # make sure that value is actually a list
+        if not type(value).__name__ == 'list':
+            return False
+
+        # iterate over each element of the list
+        # and check whether it matches T
+        list_element_type = re.search(r"^List\[(\w+)\]$", expected_type).group(1)
```

Comment:
Isn't homogeneity of the collection ensured if its type is `list`? Probably this check is redundant?

Reply:
Sadly no:
```
>>> l = [1, 1.0, 'foo']
>>> type(l).__name__
'list'
```

Reply:
Wow! Though not so surprising given that Python doesn't have a "real" type system...